### PR TITLE
FEAT: Remove selected cache models

### PR DESCRIPTION
**Pull request by hainaweiben**
Bash:
You can use the command `xinference remove-cache --help` to learn how to delete cached models.

### CODE REVIEW

**Comment by amumu96** [comment]
Could you add some tests?

File: xinference/core/cache_tracker.py [hunk 1]
```diff
@@ -111,3 +113,56 @@ def list_cached_models(self) -> List[Dict[Any, Any]]:
...
+
+    def get_remove_cached_models(
+        self, model_name: str, checked: bool = False
+    ) -> Dict[str, Dict[str, str]]:
+        assert checked == False
+        model_file_location = {}
+        for model, model_versions in self._model_name_to_version_info.items():
+            if model_name.lower() == model.lower():
+                for version_info in model_versions:
+                    if version_info["cache_status"]:
```
Comment (amumu96) [inline xinference/core/cache_tracker.py:125]: `get()` may be better

File: xinference/core/cache_tracker.py [hunk 2]
```diff
@@ -111,3 +113,56 @@ def list_cached_models(self) -> List[Dict[Any, Any]]:
...
+                        model_file_location = version_info["model_file_location"]
```
Comment (amumu96) [inline xinference/core/cache_tracker.py:126]: ibid

File: xinference/core/cache_tracker.py [hunk 3]
```diff
@@ -111,3 +113,56 @@ def list_cached_models(self) -> List[Dict[Any, Any]]:
...
+
+        return {model_name: model_file_location}
+
+    def remove_cached_models(
+        self, model_name: str, model_file_location: Dict[str, str]
+    ) -> str:
+        for model, model_versions in self._model_name_to_version_info.items():
+            if model_name.lower() == model.lower():
+                for version_info in model_versions:
+                    if version_info["model_file_location"] == model_file_location:
```
Comment (amumu96) [inline xinference/core/cache_tracker.py:136]: You know, `get`

File: xinference/core/cache_tracker.py [hunk 4]
```diff
@@ -111,3 +113,56 @@ def list_cached_models(self) -> List[Dict[Any, Any]]:
...
+        model_file_location = {}
+        for model, model_versions in self._model_name_to_version_info.items():
+            if model_name.lower() == model.lower():
+                for version_info in model_versions:
+                    if version_info["cache_status"]:
+                        model_file_location = version_info["model_file_location"]
+
+        return {model_name: model_file_location}
+
+    def remove_cached_models(
```
Comment (amumu96) [inline xinference/core/cache_tracker.py:130]: `remove` should be excuted by worker, you know, `cache_tracker` was created by `supervisor` so it runs on the device where supervisor locates. if you want to remove path on worker device, `cache_tracker` will tell you `No such file or directory`

File: xinference/api/restful_api.py [hunk 5]
```diff
@@ -503,6 +503,26 @@ def serve(self, logging_conf: Optional[dict] = None):
                 else None
             ),
         )
+        self._router.add_api_route(
+            "/v1/get_remove_cached_models",
```
Comment (qinxuye) [inline xinference/api/restful_api.py:507]: The name looks weird and not so Restful.

File: xinference/api/restful_api.py [hunk 6]
```diff
@@ -503,6 +503,26 @@ def serve(self, logging_conf: Optional[dict] = None):
...
+            self.get_remove_cached_models,
+            methods=["GET"],
+            dependencies=(
+                [Security(self._auth_service, scopes=["models:list"])]
+                if self.is_authenticated()
+                else None
+            ),
+        )
+        self._router.add_api_route(
+            "/v1/remove_cached_models",
```
Comment (qinxuye) [inline xinference/api/restful_api.py:517]: Ditto.

File: xinference/api/restful_api.py [hunk 7]
```diff
@@ -1554,6 +1582,43 @@ async def get_cluster_version(self) -> JSONResponse:
             logger.error(e, exc_info=True)
             raise HTTPException(status_code=500, detail=str(e))
 
+    async def get_remove_cached_models(self, request: Request) -> JSONResponse:
```
Comment (qinxuye) [inline xinference/api/restful_api.py:1585]: The function name can be refined.

File: xinference/client/restful/restful_client.py [hunk 8]
```diff
@@ -1181,3 +1190,66 @@ def query_engine_by_model_name(self, model_name: str):
 
         response_data = response.json()
         return response_data
+
+    def get_remove_cached_models(
+        self, model_version: str, worker_ip: Optional[str] = None
+    ) -> Dict[str, Any]:
+        """
+        Get the cached models with the model name cached on the server.
```
Comment (qinxuye) [inline xinference/client/restful/restful_client.py:1198]:
```suggestion
        Get the cached models with the model path cached on the server.
```

File: xinference/client/restful/restful_client.py [hunk 9]
```diff
@@ -1181,3 +1190,66 @@ def query_engine_by_model_name(self, model_name: str):
...
+        response = requests.get(url, headers=self._headers, json=payload)
+        if response.status_code != 200:
+            raise RuntimeError(
+                f"Failed to get paths by model name, detail: {_get_error_string(response)}"
+            )
+
+        response_data = response.json()
+        return response_data
+
+    def remove_cached_models(
```
Comment (qinxuye) [inline xinference/client/restful/restful_client.py:1225]: Name could be refined.

**Pull request closed without merging**